core/templates/domain/collection/CollectionRightsObjectFactory.ts

```diff
@@ -136,10 +136,10 @@ export class CollectionRightsObjectFactory {
   createEmptyCollectionRights(): CollectionRights {
     return new CollectionRights({
       owner_names: [],
-      collection_id: null,
-      can_edit: null,
-      can_unpublish: null,
-      is_private: null
+      collection_id: 0,
+      can_edit: false,
+      can_unpublish: false,
+      is_private: false
```

Comment:
Have you verified that this doesn't cause any bugs?

Reply:
yeahh..I tried all the front-end and end to end test. I checked the files where the given type is called it seems they always intialise it after creating it with this function as it's earlier values were null. Do let me know if there is any other type of check that I can do to cross verify it once again.

Reply:
You should check all the places where this function is called. Maybe there is some functionality that requires these values to be `null`. Plus assigning `0` as the collectionId may be wrong because there maybe a collection with the id `0`. 

If required you also may need to refactor the code in other places as well.

Reply:
This user-defined type is being initialized every time it is being called, and there is no check of it with the null value for more reference please have a look at ```collection-editor-state.service```. For collectionId code seems to be working but I think I should change it to -1 to make a distinction. Do let me know if I'm heading in the right direction.

Reply:
I think instead what we can do is remove this function. And make the following changes

Change this
https://github.com/oppia/oppia/blob/bd8f6376bb8fb837bd48332ba4f5810d59e39142/core/templates/domain/collection/CollectionRightsObjectFactory.ts#L112-L118

to

```typescript
 static copyFromCollectionRights(otherCollectionRights: CollectionRights): CollectionRights {
    return new CollectionRights({
      collection_id: otherCollectionRights.getCollectionId(),
      can_edit: otherCollectionRights.canEdit(),
      can_unpublish: otherCollectionRights.canUnpublish(),
      is_private: otherCollectionRights.isPrivate(),
      owner_names: otherCollectionRights.getOwnerNames()
    });
  }
```

Change this
https://github.com/oppia/oppia/blob/bd8f6376bb8fb837bd48332ba4f5810d59e39142/core/templates/pages/collection-editor-page/services/collection-editor-state.service.ts#L41-L42

to

```typescript
var _collectionRights;
```

Change this

https://github.com/oppia/oppia/blob/bd8f6376bb8fb837bd48332ba4f5810d59e39142/core/templates/pages/collection-editor-page/services/collection-editor-state.service.ts#L61

to 

```typescript
_collectionRights = CollectionRights.copyFromCollectionRights(collectionRights)
```

/cc @kevintab95 wdyt?

Reply:
Sure will look into this direction @nishantwrp . Thanks!

Reply:
@nishantwrp while making the function as suggested by you and making other files to work accordingly. I have to add this line ```import { CollectionRights } from 'domain/collection/CollectionRightsObjectFactory';```.(I tried without it as well but it wasn't working).
 Now in the files there is already a call with
```import { CollectionRightsObjectFactory} from 'domain/collection/CollectionRightsObjectFactory'; ``` 
 or with ```require('domain/collection/CollectionObjectFactory.ts');```. 
So I wanted to know that is there an alternative way to use the ```CollectionRights```  using the already present declaration?

Reply:
Yup, sg!

Reply:
@arpit1912 Yeah, you can add a `import { CollectionRights } from 'domain/collection/CollectionRightsObjectFactory';` statement even if there's a require statement. Thanks!

Reply:
@nishantwrp  while making the above changes I'm facing the error shown below:
![Screenshot from 2020-09-26 23-37-39](https://user-images.githubusercontent.com/46648172/94348286-767da580-0058-11eb-91dd-339949971636.jpg)

The error is generated from the given below test:
![Screenshot from 2020-09-27 00-24-49](https://user-images.githubusercontent.com/46648172/94348289-7aa9c300-0058-11eb-95f6-29c71cd6747e.jpg)

I tried some checks on it, seems like the ```previousCollectionRights``` is getting initialized every time due to the static function while the test was expecting it to be undefined. Please let me know how I should tackle it.

Reply:
Seeing the error, I think one of the `expectedCollectionRights` and `previousCollectionRights` is `undefined`.

Reply:
its the ```previousCollectionRights``` which should be ```undefined```.

Reply:
It should be undefined or it is undefined? Actually lets discuss this on hangouts for faster discussion

Reply:
@nishantwrp  as per me it should be undefined.. also I have sent you a hangouts request. Thanks!

Reply:
@arpit1912, I have not received any request. My email is mittalnishant14@gmail.com